### PR TITLE
test: Fix integration tests for new identity

### PR DESCRIPTION
**Pull request by stepansergeevitch**
Fixed integration tests issues, that appeared in new identity. Removed some existing system engine tests, since they've overlapped with resource manager tests.
Successful integration tests run: https://github.com/firebolt-db/firebolt-python-sdk/actions/runs/5642758144

### CODE REVIEW

File: tests/integration/dbapi/async/test_errors_async.py [hunk 1]
```diff
@@ -92,7 +92,8 @@ async def test_database_not_exists(
             await connection.cursor().execute("show tables")
 
     assert (
-        str(exc_info.value) == f"Database {new_db_name} does not exist"
+        str(exc_info.value)
+        == f"Engine {engine_name} is not attached to {new_db_name}, but to {database_name}"
```
Comment (alexradzin) [inline tests/integration/dbapi/async/test_errors_async.py:96]: IMHO message like "Engine {engine_name} is  attached to {database_name} instead  of {new_db_name}" is clearer.

**Review by alexradzin — COMMENTED** [review]
I want to approve this PR because I do not understand it  anyway. Unfortunately I cannot choose "approve", probably it is because I have added comments.

 I have added a couple of questions and will appreciate if you can explain me. Also please take a look on 3 "code smells" found by Sonar.

File: tests/integration/dbapi/async/test_queries_async.py [hunk 2]
```diff
@@ -127,10 +127,10 @@ async def test_long_query(
 async def test_drop_create(connection: Connection) -> None:
     """Create and drop table/index queries are handled properly."""
 
-    async def test_query(c: Cursor, query: str, empty_response=True) -> None:
+    async def test_query(c: Cursor, query: str) -> None:
```
[the remark this answers is not in the document]
Comment (stepansergeevitch) [reply tests/integration/dbapi/async/test_queries_async.py:130]: Yes, we always receive 0 now. We used to receive 0 or -1 in different cases

File: tests/integration/dbapi/async/test_errors_async.py (hunk 1 above)
Comment (stepansergeevitch) [reply tests/integration/dbapi/async/test_errors_async.py:96]: Sounds good

**Review by alexradzin — APPROVED** [review]

**Comment by stepansergeevitch** [comment]
Integration tests are passing: https://github.com/firebolt-db/firebolt-python-sdk/actions/runs/5645658069/job/15316646593

**Pull request merged**